City-of-Helsinki/drupal-infofinland
# UHF-9631: theme updates

# [UHF-9631](https://helsinkisolutionoffice.atlassian.net/browse/UHF-9631)


## What was done


Changes made after 6.11.2023 were reverted. This pull requests aims to replace these pull requests:

- https://github.com/City-of-Helsinki/drupal-infofinland/pull/196 17.11.
- https://github.com/City-of-Helsinki/drupal-infofinland/pull/197 17.11.
- https://github.com/City-of-Helsinki/drupal-infofinland/pull/198 17.11.
- https://github.com/City-of-Helsinki/drupal-infofinland/pull/199 17.11.
- https://github.com/City-of-Helsinki/drupal-infofinland/pull/200 17.11.
- https://github.com/City-of-Helsinki/drupal-infofinland/pull/203 6.11.
- https://github.com/City-of-Helsinki/drupal-infofinland/pull/209 12.12.
- https://github.com/City-of-Helsinki/drupal-infofinland/pull/207 12.12.
- https://github.com/City-of-Helsinki/drupal-infofinland/pull/208 12.12.
- https://github.com/City-of-Helsinki/drupal-infofinland/pull/209 12.12.
- https://github.com/City-of-Helsinki/drupal-infofinland/pull/210 12.12.
- https://github.com/City-of-Helsinki/drupal-infofinland/pull/211 12.12.
- https://github.com/City-of-Helsinki/drupal-infofinland/pull/215 12.12.
- https://github.com/City-of-Helsinki/drupal-infofinland/pull/220 12.12.
- https://github.com/City-of-Helsinki/drupal-infofinland/pull/221 12.12.
- https://github.com/City-of-Helsinki/drupal-infofinland/pull/222 13.12.
- https://github.com/City-of-Helsinki/drupal-infofinland/pull/223 14.12.
- https://github.com/City-of-Helsinki/drupal-infofinland/pull/224 14.12.

Instead of updating all the outdated dependencies, it was easier to update the whole theme. I copied all files from `drupal-helfi-platform` subtheme to infofinland theme and manually checked which changes are sensible.

I also removed `hdbt_subtheme` since Snyk seems to generate update pull requests against its dependencies and they are are just unnecessary noise since the theme is not used.

## How to install

* Make sure your instance is up and running on correct branch.
  * `git checkout UHF-9631-js-changes`
  * `make fresh`
* Run `make drush-cr`

## How to test


- Run `npm audit`. It should find 0 vulnerabilities.
- Check that infofinland Drupal theme works.
- Make sure code follows our standards


[UHF-9631]: https://helsinkisolutionoffice.atlassian.net/browse/UHF-9631?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
